### ISSUE TITLE
False Positive (unused-arguement)

### ISSUE THREAD

**Issue by mohi7solanki** [PyCQA/pylint#3074]
### Steps to reproduce
```
def request_retry_sync(retries=3, cooldown=1):
    def wrap(func):
        @wraps(func)
        def innner(*args, **kwargs):
            nonlocal cooldown
            for retry_count in range(retries+1):
                response = func(*args, **kwargs)
                try:
                    response.raise_for_status()
                except (
                    requests.exceptions.ConnectionError,
                    requests.exceptions.Timeout,
                    requests.exceptions.HTTPError,
                ) as err:
                    if retry_count == retries:
                        # last retry raise the exception
                        raise
                    time.sleep(cooldown)
                    cooldown *= 2
                    LOGGER.warning(
                        '%s during %s execution %s of "%s" retries attempted',
                        err.__class__.__name__,
                        func.__name__,
                        retry_count + 1,
                        retries
                    )
                else:
                    return response
        return innner
    return wrap
```

### Current behavior
pylint raising unused arguement for `cooldown`

### Expected behavior
cooldown is used. should not raise the error

### pylint --version output
pylint 2.3.1
astroid 2.2.5
Python 3.7.4 (default, Jul 16 2019, 07:12:58) 
[GCC 9.1.0]

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by jacobtylerwalls** [comment]
Duplicate of #5187 (more minimal example)

**Comment by jacobtylerwalls** [comment]
Duplicate of #5187 (more minimal example)
